pinot-query-planner/src/main/java/org/apache/pinot/query/planner/nodes/TableScanNode.java

```diff
@@ -22,8 +22,12 @@
 
 
 public class TableScanNode extends AbstractStageNode {
-  private final String _tableName;
-  private final List<String> _tableScanColumns;
+  private String _tableName;
+  private List<String> _tableScanColumns;
+
+  public TableScanNode(int stageId) {
```

Comment:
These constructors are probably confusing. Why do we need this ?

Reply:
we need to ensure that the StageNode are actually referred by a stageId. a StageNode without a proper stageId associated is not a valid stagenode